### PR TITLE
refactor(CDM): refactor to new SDK

### LINKED ISSUE #1502: [CDM] refactor resource_huaweicloud_cdm_cluster_v1 to new SDK
please refactor resource_huaweicloud_cdm_cluster_v1 to new SDK

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
1. refactor to new SDK
2. add some exported  attributes: private_ip,manage_ip, public_endpoint,status

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1502 

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cdm' TESTARGS='-run=TestAccResourceCdmCluster_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cdm -v -run=TestAccResourceCdmCluster_basic -timeout 360m -parallel 4
=== RUN   TestAccResourceCdmCluster_basic
=== PAUSE TestAccResourceCdmCluster_basic
=== CONT  TestAccResourceCdmCluster_basic
--- PASS: TestAccResourceCdmCluster_basic (929.31s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cdm       929.523s
```

```
make testacc TEST='./huaweicloud/services/acceptance/cdm' TESTARGS='-run=TestAccResourceCdmCluster_all'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cdm -v -run=TestAccResourceCdmCluster_all -timeout 360m -parallel 4
=== RUN   TestAccResourceCdmCluster_all
=== PAUSE TestAccResourceCdmCluster_all
=== CONT  TestAccResourceCdmCluster_all
--- PASS: TestAccResourceCdmCluster_all (1410.96s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cdm       1411.105s
```

### CODE REVIEW

File: docs/resources/cdm_cluster.md [hunk 1]
```diff
@@ -4,26 +4,29 @@ subcategory: "Cloud Data Migration (CDM)"
...
-  security_group_id = huaweicloud_networking_secgroup.secgroup.id
-  subnet_id         = "{{ network_id }}"
-  vpc_id            = "{{ vpc_id }}"
-  version           = "{{ version }}"
+  name              = var.name
+  availability_zone = var.availability_zone
+  flavor_id         = data.huaweicloud_cdm_flavors.test.flavors[0].id
+  subnet_id         = var.subnet_id
+  vpc_id            = var.vpc_id
+  security_group_id = var.secgroup_id
```
Comment (chengxiangdong) [inline docs/resources/cdm_cluster.md:29]: missing required argument: `version`

File: docs/resources/cdm_cluster.md [hunk 2]
```diff
@@ -32,69 +35,112 @@ resource "huaweicloud_cdm_cluster" "cluster" {
 The following arguments are supported:
 
 * `region` - (Optional, String, ForceNew) The region in which to create the cluster resource. If omitted, the
-  provider-level region will be used. Changing this creates a new CDM cluster resource.
+  provider-level region will be used. Changing this parameter will creates a new resource.
```
Comment (chengxiangdong) [inline docs/resources/cdm_cluster.md]: s/creates/create

File: docs/resources/cdm_cluster.md [hunk 3]
```diff
@@ -32,69 +35,112 @@ resource "huaweicloud_cdm_cluster" "cluster" {
...
+ Changing this parameter will create a new resource.
 
-* `email` - (Optional, List, ForceNew) Notification email addresses. The max number is 5. Changing this parameter will
-  create a new resource.
+* `enterprise_project_id` - (Optional, String, ForceNew) Specifies the enterprise project id.
+ Changing this parameter will create a new resource.
 
-* `enterprise_project_id` - (Optional, String, ForceNew) The enterprise project id. Changing this parameter will create
-  a new resource.
+* `is_auto_off` - (Optional, Bool, ForceNew) Specifies Whether to enable auto shutdown. The auto shutdown and scheduled
```
Comment (chengxiangdong) [inline docs/resources/cdm_cluster.md:60]: add description that default value is false

File: docs/resources/cdm_cluster.md [hunk 4]
```diff
@@ -32,69 +35,112 @@ resource "huaweicloud_cdm_cluster" "cluster" {
...
+ startup/shutdown functions cannot be enabled at the same time. When auto shutdown is enabled, if no job is running in
+  the cluster and no scheduled job is created, a cluster will be automatically shut down 15 minutes after it starts
+   running to reduce costs. Changing this parameter will create a new resource.
 
-* `is_auto_off` - (Optional, Bool, ForceNew) Whether to automatically shut down. Changing this parameter will create a
-  new resource.
+* `schedule_boot_time` - (Optional, String, ForceNew) Specifies time for scheduled startup of a CDM cluster.
```
Comment (chengxiangdong) [inline docs/resources/cdm_cluster.md:65]: `schedule_boot_time` is not very good, but the published ones do not need to be updated.
It is recommended to add a description of the time format.

File: docs/resources/cdm_cluster.md [hunk 5]
```diff
@@ -32,69 +35,112 @@ resource "huaweicloud_cdm_cluster" "cluster" {
...
+ The CDM cluster starts at this time every day. The scheduled startup/shutdown and auto shutdown function cannot be
+  enabled at the same time. Changing this parameter will create a new resource.
 
-* `phone_num` - (Optional, List, ForceNew) Notification phone numbers. The max number is 5. Changing this parameter will
-  create a new resource.
+* `schedule_off_time` - (Optional, String, ForceNew) Specifies time for scheduled shutdown of a CDM cluster.
```
Comment (chengxiangdong) [inline docs/resources/cdm_cluster.md:69]: ditto

File: huaweicloud/services/cdm/resource_huaweicloud_cdm_cluster.go [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.CdmV11Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CDM v1 client, err=%s", err)
+	}
+
+	detail, dErr := clusters.Get(client, d.Id())
+	if dErr != nil {
+		return fmtp.DiagErrorf("Error query DLI CDM cluster %q:%s", d.Id(), dErr)
```
Comment (chengxiangdong) [inline huaweicloud/services/cdm/resource_huaweicloud_cdm_cluster.go]: pls use `common.CheckDeletedDiag`

File: huaweicloud/services/cdm/resource_huaweicloud_cdm_cluster.go [hunk 7]
```diff
@@ -0,0 +1,423 @@
...
+		setInstancesToState(d, detail.Instances),
+		d.Set("schedule_boot_time", detail.ScheduleBootTime),
+		d.Set("schedule_off_time", detail.ScheduleOffTime),
+		d.Set("created", detail.Created),
+		d.Set("public_ip", detail.Instances[0].PublicIp),
+		d.Set("public_endpoint", detail.PublicEndpoint),
+		d.Set("status", detail.StatusDetail),
+	)
+	if setSdErr := mErr.ErrorOrNil(); setSdErr != nil {
+		return fmtp.DiagErrorf("Error setting vault fields: %s", setSdErr)
```
Comment (chengxiangdong) [inline huaweicloud/services/cdm/resource_huaweicloud_cdm_cluster.go]: s/vault fields/CDM attributes
the following code is clearer
```
if mErr.ErrorOrNil() != nil {
    return fmtp.DiagErrorf("Error setting vault fields: %s", mErr)
}
```

File: huaweicloud/services/cdm/resource_huaweicloud_cdm_cluster.go [hunk 8]
```diff
@@ -0,0 +1,423 @@
...
+	if err != nil {
+		return diag.Errorf("error creating CDM v1 client, err=%s", err)
+	}
+
+	dErr := clusters.Delete(client, d.Id(), clusters.ClusterDeleteOpts{})
+	if dErr.Err != nil {
+		return fmtp.DiagErrorf("delete CDM cluster failed. %q:%s", d.Id(), dErr)
+	}
+
+	dCreateErr := checkClusterDeleteResult(ctx, client, d.Id(), d.Timeout(schema.TimeoutDelete))
```
Comment (chengxiangdong) [inline huaweicloud/services/cdm/resource_huaweicloud_cdm_cluster.go]: What does `dCreateErr` mean？It is confusing.

File: huaweicloud/services/cdm/resource_huaweicloud_cdm_cluster.go [hunk 9]
```diff
@@ -0,0 +1,423 @@
...
+	if dCreateErr != nil {
+		return diag.FromErr(dCreateErr)
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func checkClusterCreateResult(ctx context.Context, client *golangsdk.ServiceClient, id string,
```
Comment (chengxiangdong) [inline huaweicloud/services/cdm/resource_huaweicloud_cdm_cluster.go]: suggest use `waitForClusterCreated`

File: huaweicloud/services/cdm/resource_huaweicloud_cdm_cluster.go [hunk 10]
```diff
@@ -0,0 +1,423 @@
...
+		PollInterval: 20 * timeout,
+		Delay:        20 * time.Second,
+	}
+	_, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmtp.Errorf("error waiting for CDM cluster (%s) to be created: %s", id, err)
+	}
+	return nil
+}
+func checkClusterDeleteResult(ctx context.Context, client *golangsdk.ServiceClient, id string,
```
Comment (chengxiangdong) [inline huaweicloud/services/cdm/resource_huaweicloud_cdm_cluster.go]: suggest use `waitForClusterDeleted`

File: docs/resources/cdm_cluster.md [hunk 11]
```diff
@@ -32,69 +35,112 @@ resource "huaweicloud_cdm_cluster" "cluster" {
...
+ You can then decide if changes should be applied to the cluster, or the resource definition
+should be updated to align with the cluster. Also you can ignore changes as below.
+
+```
+resource "huaweicloud_cdm_cluster" "test" {
+    ...
+
+  lifecycle {
+    ignore_changes = [
+      email, node_admin_pass, phone_num,
```
Comment (chengxiangdong) [inline docs/resources/cdm_cluster.md]: pls remove `node_admin_pass`

File: docs/resources/cdm_cluster.md (hunk 1 above)
Comment (hmbig2) [reply docs/resources/cdm_cluster.md:29]: it's Optional now

File: huaweicloud/services/cdm/resource_huaweicloud_cdm_cluster.go [hunk 12]
```diff
@@ -0,0 +1,425 @@
...
+		d.Set("schedule_boot_time", detail.ScheduleBootTime),
+		d.Set("schedule_off_time", detail.ScheduleOffTime),
+		d.Set("created", detail.Created),
+		d.Set("public_ip", detail.Instances[0].PublicIp),
+		d.Set("public_endpoint", detail.PublicEndpoint),
+		d.Set("status", detail.StatusDetail),
+	)
+
+	if mErr.ErrorOrNil() != nil {
+		return fmtp.DiagErrorf("Error setting vault fields: %s", mErr)
```
Comment (chengxiangdong) [inline huaweicloud/services/cdm/resource_huaweicloud_cdm_cluster.go]: s/vault/CDM

**Pull request merged**